File: aladdinworks4_ui/src/services/SwitchDeviceService.js
import http from "../http-common"; 

class SwitchDeviceService {
  getAllSwitchDevices(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/switchDevice/switchDevices`, searchDTO);
  }

  get(switchDeviceId) {
    return this.getRequest(`/switchDevice/${switchDeviceId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/switchDevice?field=${matchData}`, null);
  }

  addSwitchDevice(data) {
    return http.post("/switchDevice/addSwitchDevice", data);
  }

  update(data) {
  	return http.post("/switchDevice/updateSwitchDevice", data);
  }
  
  uploadImage(data,switchDeviceId) {
  	return http.postForm("/switchDevice/uploadImage/"+switchDeviceId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SwitchDeviceService();

File: src/main/java/com/aladdinworks4/service/SwitchDeviceService.java
package com.aladdinworks4.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks4.domain.SwitchDevice;
import com.aladdinworks4.dto.SwitchDeviceDTO;
import com.aladdinworks4.dto.SwitchDeviceSearchDTO;
import com.aladdinworks4.dto.SwitchDevicePageDTO;
import com.aladdinworks4.dto.SwitchDeviceConvertCriteriaDTO;
import com.aladdinworks4.service.GenericService;
import com.aladdinworks4.dto.common.RequestDTO;
import com.aladdinworks4.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SwitchDeviceService extends GenericService<SwitchDevice, Integer> {

	List<SwitchDevice> findAll();

	ResultDTO addSwitchDevice(SwitchDeviceDTO switchDeviceDTO, RequestDTO requestDTO);

	ResultDTO updateSwitchDevice(SwitchDeviceDTO switchDeviceDTO, RequestDTO requestDTO);

    Page<SwitchDevice> getAllSwitchDevices(Pageable pageable);

    Page<SwitchDevice> getAllSwitchDevices(Specification<SwitchDevice> spec, Pageable pageable);

	ResponseEntity<SwitchDevicePageDTO> getSwitchDevices(SwitchDeviceSearchDTO switchDeviceSearchDTO);
	
	List<SwitchDeviceDTO> convertSwitchDevicesToSwitchDeviceDTOs(List<SwitchDevice> switchDevices, SwitchDeviceConvertCriteriaDTO convertCriteria);

	SwitchDeviceDTO getSwitchDeviceDTOById(Integer switchDeviceId);







}





